packages/banner/src/BannerAnimator/BannerAnimator.js

```diff
@@ -0,0 +1,204 @@
+    /* eslint-disable react/no-unused-prop-types */
+    /** Determines the visibility of the banner */
+    isVisible: PropTypes.bool.isRequired,
+    /** Determines the type of animation used */
+    hasBounce: PropTypes.bool.isRequired,
+    /** Determines the type of animation used */
+    hasPush: PropTypes.bool.isRequired,
+    /** Determines the direction of the animation */
+    position: PropTypes.oneOf(AVAILABLE_POSITIONS),
+    /* eslint-enable react/no-unused-prop-types */
+    /** A render prop, that renders the component to be animated */
+    children: PropTypes.func.isRequired
+  };
+
+  /** @type {BannerAnimatorState} */
+  state = {};
+
+  /**
+   * @param {BannerAnimatorProps} prevProps
+   * @param {BannerAnimatorState} prevState
+   */
+  componentDidUpdate(prevProps, prevState) {
+    const { status: prevStatus } = prevState;
+    const { status } = this.state;
+
+    if (prevStatus === statuses.EXPANDED && status === statuses.COLLAPSING) {
+      this.collapseFromExpanded();
+      return;
+    }
+    if (prevStatus === statuses.COLLAPSING && status === statuses.EXPANDING) {
+      this.expand();
+      return;
+    }
+    if (prevStatus === statuses.EXPANDING && status === statuses.COLLAPSING) {
+      this.collapse();
+    }
+  }
+
+  expand() {
+    window.requestAnimationFrame(() => {
+      this.setState(animateExpand);
+    });
+  }
+
+  collapse() {
+    window.requestAnimationFrame(() => {
+      this.setState(animateCollapse);
+    });
+  }
+
+  collapseFromExpanded() {
+    window.requestAnimationFrame(() => {
+      this.setState(prepareCollapse, () => {
+        this.collapse();
+      });
+    });
+  }
+
+  /** @type {BannerAnimatorProps} */
+  props;
+
+  handleReady = () => {
+    const { status } = this.state;
+
+    if (status === statuses.EXPANDING) {
+      this.expand();
+    }
+  };
+
+  /** @param {TransitionEvent} event */
+  handleTransitionEnd = event => {
+    const { innerWrapper } = this.state;
+
+    if (event.target !== innerWrapper) return;
+
+    const { status } = this.state;
+
+    if (status === statuses.COLLAPSING) {
+      window.requestAnimationFrame(() => {
+        this.setState(endCollapse);
+      });
+      return;
+    }
+
+    if (status === statuses.EXPANDING) {
+      window.requestAnimationFrame(() => {
+        this.setState(endExpand);
+      });
+    }
+  };
+
+  /** @param {HTMLDivElement} wrapper */
+  refWrapper = wrapper => {
+    this.setState({ wrapper });
+  };
+
+  /** @param {HTMLDivElement} innerWrapper */
+  refInnerWrapper = innerWrapper => {
+    this.setState({ innerWrapper });
+  };
+
+  render() {
+    const { children: renderChildren } = this.props;
+    const { status, wrapperStyle, innerWrapperStyle } = this.state;
+    const {
+      handleReady,
+      handleTransitionEnd,
+      refInnerWrapper,
+      refWrapper
+    } = this;
+    const children =
+      status === statuses.COLLAPSED ? null : renderChildren({ handleReady });
+    const isBusy =
```

Comment:
We should increase our max line length. Currently it's only hindering readability.